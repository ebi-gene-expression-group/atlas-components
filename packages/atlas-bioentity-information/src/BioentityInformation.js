import React from 'react'
import PropTypes from 'prop-types'

import BioentityProperty from './BioentityProperty';

class BioentityInformation extends React.Component {

    constructor(props) {
        super(props)
    }

  render() {
        return (
          <div className={`row row-padding small-12`}>
              <ul className={`accordion`} data-accordion data-allow-all-closed="true">
                  <li className={`accordion-item`} data-accordion-item>
                      <a href="#geneInfo" className={`accordion-title`} style={{fontSize: '1.1rem'}}>Gene Information</a>
                      <div id="geneInfo" className={`accordion-content`} data-tab-content>

                          <table>
                              <tbody>
                              {this.props.bioentityProperties.map(function(bioentityProperty){
                                  return (
                                      <BioentityProperty
                                          key={bioentityProperty.type}
                                          {...bioentityProperty} />
                                  )
                              })}
                              </tbody>
                          </table>

                      </div>
                  </li>
              </ul>
          </div>
    )
  }
}

BioentityInformation.propTypes = {
    atlasUrl: PropTypes.string,
    bioentityProperties: PropTypes.array
}

export default BioentityInformation
