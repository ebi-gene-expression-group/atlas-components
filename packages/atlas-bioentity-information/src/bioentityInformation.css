.gxaBioentityInformationCard {
    margin-top: 20px;
}

.gxaBioentityInformationCard table {
    margin: 0;
    width: auto;
    border: none;
}

.gxaBioentityInformationCard td {
    border: none;
}


td.gxaBioentityInformationCardPropertyType {
    font-size: 14px;
    font-weight: bold;
    white-space: nowrap;
    padding-right: 2em;
    border:none;
}

.gxaBioentityInformationCardPropertyValue {
    border: none;
}

/*ul#geneInformation>li a.gxaAccordionTitle {*/
    /*font-size: 1.2rem !important;*/
/*}*/
