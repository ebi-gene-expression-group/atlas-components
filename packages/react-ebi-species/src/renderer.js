"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
