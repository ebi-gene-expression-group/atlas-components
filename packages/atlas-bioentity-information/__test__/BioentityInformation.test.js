import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import BioentityInformation from '../src/BioentityInformation.js'

Enzyme.configure({ adapter: new Adapter() })

const bioentityData = [{
    "type": "go",
    "name": "Gene Ontology",
    "values": [

        {
            "text": "Link of relevance 1",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0000978",
            "relevance": 1
        },
        {
            "text": "a Link of relevance 2",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0000978",
            "relevance": 2
        },
        {
            "text": "Link of relevance 3",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0000978",
            "relevance": 3
        },
        {
            "text": "b Link of relevance 2",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0000978",
            "relevance": 2
        },
        {
            "text": "Link of relevance 3- another",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0000978",
            "relevance": 3
        },
        {
            "text": "RNA polymerase II core promoter proximal region sequence-specific DNA binding",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0000978",
            "relevance": 0
        }, {
            "text": "nucleic acid binding",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0003676",
            "relevance": 0
        }, {
            "text": "intracellular",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0005622",
            "relevance": 0
        }, {
            "text": "nucleus",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0005634",
            "relevance": 0
        }, {
            "text": "transcription, DNA-templated",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0006351",
            "relevance": 0
        }, {
            "text": "regulation of transcription, DNA-templated",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0006355",
            "relevance": 0
        }, {
            "text": "zinc ion binding",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0008270",
            "relevance": 0
        }, {
            "text": "metal ion binding",
            "url": "http://amigo.geneontology.org/amigo/term/GO%3A0046872",
            "relevance": 0
        }]
    }, {
    "type": "pathway",
    "name": "Reactome Pathways",
    "values": []
}];

describe(`BioentityInformation`, () => {
  test(`should render without throwing an error`, () => {
      expect(mount(<BioentityInformation bioentityProperties={bioentityData} />)).toHaveLength(1)
  })

  test('render props lenght', () => {
      const wrapper = shallow(<BioentityInformation bioentityProperties={bioentityData}/>);
      expect(wrapper.find('BioentityProperty')).toHaveLength(2)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<BioentityInformation bioentityProperties={bioentityData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
