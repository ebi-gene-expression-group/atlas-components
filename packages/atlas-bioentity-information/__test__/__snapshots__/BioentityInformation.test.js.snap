// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BioentityInformation matches snapshot 1`] = `
<div
  className="row row-padding small-12"
>
  <ul
    className="accordion"
    data-accordion={true}
    data-allow-all-closed="true"
  >
    <li
      className="accordion-item"
      data-accordion-item={true}
    >
      <a
        className="accordion-title"
        href="#geneInfo"
        style={
          Object {
            "fontSize": "1.1rem",
          }
        }
      >
        Gene Information
      </a>
      <div
        className="accordion-content"
        data-tab-content={true}
        id="geneInfo"
      >
        <table>
          <tbody>
            <tr>
              <td
                className="gxaBioentityInformationCardPropertyType"
              >
                Gene Ontology
              </td>
              <td>
                <div>
                  <span>
                    <a
                      className="bioEntityCardLink"
                      href="http://amigo.geneontology.org/amigo/term/GO%3A0000978"
                      target="_blank"
                    >
                      Link of relevance 3
                    </a>
                    <span>
                      , 
                    </span>
                    <a
                      className="bioEntityCardLink"
                      href="http://amigo.geneontology.org/amigo/term/GO%3A0000978"
                      target="_blank"
                    >
                      Link of relevance 3- another
                    </a>
                    <span>
                      , 
                    </span>
                    <a
                      className="bioEntityCardLink"
                      href="http://amigo.geneontology.org/amigo/term/GO%3A0000978"
                      target="_blank"
                    >
                      a Link of relevance 2
                    </a>
                    <span>
                      , 
                    </span>
                    <a
                      className="bioEntityCardLink"
                      href="http://amigo.geneontology.org/amigo/term/GO%3A0000978"
                      target="_blank"
                    >
                      b Link of relevance 2
                    </a>
                    <a
                      onClick={[Function]}
                      role="button"
                      style={
                        Object {
                          "cursor": "pointer",
                        }
                      }
                    >
                       â€¦ and 9 more
                    </a>
                  </span>
                </div>
              </td>
            </tr>
            <tr>
              <td
                className="gxaBioentityInformationCardPropertyType"
              >
                Reactome Pathways
              </td>
              <td>
                <div />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </li>
  </ul>
</div>
`;
